be/src/runtime/load_channel.cpp

```diff
@@ -21,101 +21,83 @@
 
 #include "runtime/load_channel.h"
 
+#include <memory>
+
+#include "common/closure_guard.h"
+#include "runtime/load_channel_mgr.h"
 #include "runtime/mem_tracker.h"
 #include "runtime/tablets_channel.h"
 #include "storage/lru_cache.h"
 
 namespace starrocks {
 
-LoadChannel::LoadChannel(const UniqueId& load_id, int64_t timeout_s, std::unique_ptr<MemTracker> mem_tracker)
-        : _load_id(load_id),
+LoadChannel::LoadChannel(LoadChannelMgr* mgr, const UniqueId& load_id, int64_t timeout_s,
+                         std::unique_ptr<MemTracker> mem_tracker)
+        : _load_mgr(mgr),
+          _load_id(load_id),
           _timeout_s(timeout_s),
           _mem_tracker(std::move(mem_tracker)),
           _last_updated_time(time(nullptr)) {}
 
-LoadChannel::~LoadChannel() {
-    LOG(INFO) << "load channel mem peak usage=" << _mem_tracker->peak_consumption()
-              << ", info=" << _mem_tracker->debug_string() << ", load_id=" << _load_id;
-    _tablets_channels.clear();
-}
+LoadChannel::~LoadChannel() {}
 
-Status LoadChannel::open(const PTabletWriterOpenRequest& params) {
-    int64_t index_id = params.index_id();
-    std::shared_ptr<TabletsChannel> channel;
-    {
-        std::lock_guard<std::mutex> l(_lock);
-        auto it = _tablets_channels.find(index_id);
-        if (it != _tablets_channels.end()) {
-            channel = it->second;
-        } else {
-            // create a new tablets channel
-            TabletsChannelKey key(params.id(), index_id);
-            channel.reset(new TabletsChannel(key, _mem_tracker.get()));
-            _tablets_channels.insert({index_id, channel});
-        }
-    }
-
-    RETURN_IF_ERROR(channel->open(params));
+void LoadChannel::open(brpc::Controller* cntl, const PTabletWriterOpenRequest* request,
```

Comment:
It seems that `const PTabletWriterOpenRequest& request` is better.

Reply:
updated